### PR TITLE
cli: Improve destroy retry handling

### PR DESCRIPTION
**Pull request by ironcladlou**
Make destroy commands retry until they either succeed or are cancelled.

### CODE REVIEW

File: cmd/infra/aws/destroy_iam.go [hunk 1]
```diff
@@ -50,12 +52,27 @@ func NewDestroyIAMCommand() *cobra.Command {
...
 	}
 
 	return cmd
 }
 
+func (o *DestroyIAMOptions) Run(ctx context.Context) error {
+	return wait.PollUntil(5*time.Second, func() (bool, error) {
+		err := o.DestroyIAM(ctx)
+		if err != nil {
+			log.Error(err, "error destroying infra")
```
Comment (csrwng) [inline cmd/infra/aws/destroy_iam.go:69]: Can we make this a warning? It's confusing to see an error getting logged but the command continues.

Comment (ironcladlou) [reply cmd/infra/aws/destroy_iam.go:69]: Yeah, will wait for e2e to finish

**Pull request merged**